client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -118,10 +123,24 @@ public <K, V, C> ShuffleHandle registerShuffle(
     String appId = SparkUtils.appUniqueId(dependency.rdd().context());
     initializeLifecycleManager(appId);
 
+    if (!registeredApps.contains(appId)) {
+      registeredApps.add(appId);
+      lifecycleManager.applicationCount().increment();
```

Comment:
`lifecycleManager.applicationCount().increment()` can move to `initializeLifecycleManager`